contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -507,6 +548,56 @@ contract FeeSharingCollector is
         validFromCheckpointParam(_fromCheckpoint, msg.sender, RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT)
         nonReentrant
     {
+        _withdrawRbtcTokenStartingFromCheckpoint(
+            address(0),
+            _fromCheckpoint,
+            _maxCheckpoints,
+            _receiver
+        );
+    }
+
+    /**
+     * @dev Withdraw all of the RBTC balance based starting from a specific checkpoint
+     * The function was designed to skip checkpoints with no fees for users
+     *
+     * This function will withdraw RBTC balance which is passed as _token param, so it could be either of these:
+     * - rbtc balance or
+     * - wrbtc balance which will be unwrapped to rbtc or
+     * - iwrbtc balance which will be unwrapped to rbtc or
+     *
+     * @dev WARNING! This function skips all the checkpoints before '_fromCheckpoint' irreversibly, use with care
+     *
+     * @param _token either RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT or wrbtc address or iwrbtc address
+     * @param _fromCheckpoint Skips all the checkpoints before '_fromCheckpoint'
+     *        should be calculated offchain with getNextPositiveUserCheckpoint function
+     * @param _maxCheckpoints  Maximum number of checkpoints to be processed to workaround block gas limit
+     * @param _receiver An optional tokens receiver (msg.sender used if 0)
+     */
+    function withdrawRbtcTokenStartingFromCheckpoint(
+        address _token,
+        uint256 _fromCheckpoint,
+        uint32 _maxCheckpoints,
+        address _receiver
+    ) external validFromCheckpointParam(_fromCheckpoint, msg.sender, _token) nonReentrant {
```

Comment:
here you're checking if the checkpoint is valid for `_token` , but `_withdrawRbtcTokenStartingFromCheckpoint` always updates the  `processedCheckpoints` of `RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT` . This means not only that the `_fromCheckpoint` is ignored for `_token` but also that the `processedCheckpoints` of `RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT` is updated incorrectly, which could lead to a money loss for the user. 

You need to update `_withdrawRbtcTokenStartingFromCheckpoint` as well. it needs to update the `processedCheckpoints` of `token`

Reply:
Updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/507/commits/a58adb962b3b074684f108190985bcc01a12b741)